open-apparel-registry/open-apparel-registry
# Add searchability to embed config

## Overview

Edit embed field config to allow marking a field as searchable and
add model and API support for persistence.

Connects #1533 

## Demo

<img width="552" alt="Screen Shot 2022-01-24 at 11 25 31 AM" src="https://user-images.githubusercontent.com/21046714/150825764-7a42dd29-fe20-48c6-bbbf-ab9d0c2e699e.png">
<img width="582" alt="Screen Shot 2022-01-24 at 11 57 16 AM" src="https://user-images.githubusercontent.com/21046714/150828490-474132be-2275-4c58-ad84-50a4f8535655.png">

## Notes

Adding the checkbox to the right of the fields input caused the draggability buttons to wrap to the next row on certain screen sizes. I adjusted the layout / breakpoints to move the embed code box above the fields list on smaller screen sizes. 

The new text_search_label field has been added to the embed config model, but hasn't yet been added to the embed config page. See [#1534](https://github.com/open-apparel-registry/open-apparel-registry/issues/1534) for next steps.

## Testing Instructions

* Run `./scripts/manage migrate`
* Run `./scripts/server`
* Navigate to [c2@example.com admin](http://localhost:8081/admin/api/contributor/2/change/) and login as c1@example.com. Set the embed level to `Embed Deluxe / Custom Embed` and save. 
* Logout. Login as c2@example.com and navigate to [contribute](http://localhost:6543/contribute). 
* Upload a list with extra fields and note the list ID in the url. 
[oar-extra-fields.csv](https://github.com/open-apparel-registry/open-apparel-registry/files/7927301/oar-extra-fields.csv)
* In `vagrant`, run (with the appropriate list-id):
```
./scripts/manage batch_process --list-id 16 --action parse
./scripts/manage batch_process --list-id 16 --action geocode
./scripts/manage batch_process --list-id 16 --action match
```
* Navigate to [settings](http://localhost:6543/settings) and select the Embed tab. 
* Select several fields to be 'included', then check 'searchable' for them. Refresh the page and confirm they are still included and searchable.
* Uncheck the left checkbox for a field that is searchable and note that the 'searchable' checkbox is unchecked. 
* Use 'select all' to uncheck all checkboxes and note that all searchable checkboxes are unchecked. Refresh the page to confirm persistence. 
* At various page sizes, hover over the list items to confirm that the draggable buttons render appropriately. 
* Further experiment with the embed config and field settings to confirm that the config page behaves as expected. 

## Checklist
